drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -1363,6 +1365,169 @@ impl<S: SpiServer> idl::InOrderSpRotImpl for ServerImpl<S> {
             .into()),
         }
     }
+
+    fn tq_cert(
+        &mut self,
+        _: &userlib::RecvMessage,
+        index: u32,
+        offset: u32,
+        data: idol_runtime::Leased<idol_runtime::W, [u8]>,
+    ) -> Result<(), idol_runtime::RequestError<AttestOrSprotError>> {
+        let body = ReqBody::Attest(AttestReq::TqCert {
+            index,
+            offset,
+            size: data.len() as u32,
+        });
+        let tx_size = Request::pack(&body, self.tx_buf);
+        let rsp = self
+            .do_send_recv_retries(tx_size, DUMP_TIMEOUT, DEFAULT_ATTEMPTS)
+            .map_err(AttestOrSprotError::Sprot)?;
+
+        match rsp.body {
+            Ok(RspBody::Attest(Ok(AttestRsp::TqCert))) => {
+                // Copy from the trailing data into the lease
+                if rsp.blob.len() < data.len() {
```

Comment:
This was admittedly copy pasted from elsewhere in the file and matches what everyone else is doing. I think this is "okay" because we'll just truncate the response.